common/src/PCLPointCloud2.cpp

```diff
@@ -147,9 +139,9 @@ pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPoi
   if (simple_layout)
   {
     cloud1.data.insert (cloud1.data.end (), cloud2.data.begin (), cloud2.data.end ());
-    return (true);
+    return true;
   }
-  const auto data1_size = cloud1.data.size ();
+  const std::size_t data1_size = cloud1.data.size ();
```

Comment:
Why the need to explicit `std::size_t` when auto will choose the correct type? We don't care what the `size()` function returns

Reply:
ok, I'll revert it. Simply I am not a fan of the almost-always-auto style, especially when typing the type has the same effort of typing ``auto``

Reply:
For me it's the load on the readers mind. `auto` implies don't care. Type implies there might be a implicit cast here and that care was put in explicitly adding the type. Eg: maybe the function doesn't return the same type.